clippy_lints/src/question_mark_used.rs

```diff
@@ -0,0 +1,47 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+
+use rustc_hir::{Expr, ExprKind, MatchSource};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for expressions that use the question mark operator and rejects them.
+    ///
+    /// ### Why is this bad?
+    /// Sometimes code wants to avoid the question mark operator because for instance a local
+    /// block requires a macro to re-throw errors to attach additional information to the
+    /// error.
+    ///
+    /// ### Example
+    /// ```ignore
+    /// let result = expr?;
+    /// ```
+    ///
+    /// Could be written:
+    ///
+    /// ```ignore
+    /// utility_macro!(expr);
+    /// ```
+    #[clippy::version = "pre 1.29.0"]
+    pub QUESTION_MARK_USED,
```

Comment:
I agree with @Manishearth's [comment here](https://github.com/rust-lang/rust-clippy/issues/10340#issuecomment-1428173251), there are two other lints that refer to it as question mark currently but we could rename those as a separate issue

Reply:
No I don't have a strong opinion either way on question_mark vs try, I was only proposing a change of naming from `use_of_try_operator`. I think question_mark_used is fine

Reply:
The Rust Reference also just calls it "question mark", I think try is at this point a legacy name.

Reply:
Ahh I see thanks. Looks like the book calls it question mark as well so the current name is good